crypto/secp256k1/secp256_test.go

```diff
@@ -2,6 +2,9 @@
 // Use of this source code is governed by a BSD-style license that can be found in
 // the LICENSE file.
 
+//go:build !gofuzz && cgo
+// +build !gofuzz,cgo
+
```

Comment:
Are _all_ of the tests here cgo-only? Or is it some subset? Because if the latter, then we should move those into `secp256_cgo_test.go` or something like that

Reply:
Ah, so the answer to my question is "yes". However, the tests are really in the wrong location, aren't they? 
I took one at random, `TestSignDeterministic`, moved it up a level from `secp256k1` into `crypto`: 

```golang
func TestSignDeterministic(t *testing.T) {
	seckey, err := ecdsa.GenerateKey(S256(), rand.Reader)
	if err != nil {
		panic(err)
	}
	msg := make([]byte, 32)
	copy(msg, "hi there")

	sig1, err := Sign(msg, seckey)
	if err != nil {
		t.Fatal(err)
	}
	sig2, err := Sign(msg, seckey)
	if err != nil {
		t.Fatal(err)
	}
	if !bytes.Equal(sig1, sig2) {
		t.Fatal("signatures not equal")
	}
}

```
When invoking it _here_, the test covers both the cgo implementation or nocgo, depending on flag usage.

Reply:
The package crypto/secp256k1 is inherently a cgo-based implementation. I think we could definitely move those tests, but to fix the build, we should just merge this PR.